packages/material-ui/src/TextField/TextField.d.ts

```diff
@@ -48,7 +48,10 @@ export interface BaseTextFieldProps
   /**
    * Props applied to the [`FormHelperText`](/api/form-helper-text/) element.
    */
-  FormHelperTextProps?: Partial<FormHelperTextProps>;
+  FormHelperTextProps?: FormHelperTextProps<
+    React.ElementType,
+    { component: React.ElementType | JSX.Element }
```

Comment:
What do we need `JSX.Element` for? Isn't that the actual element e.g. `<div />`?

Reply:
Isn't it for custom jsx components?

Reply:
I think this is for any value returned from createElement factory. I only use this to annotate return types of function components.

Reply:
And also how can i write test for correct use case if this is a typescript error.

Reply:
> I think this is for any value returned from createElement factory. I only use this to annotate return types of function components.

So u think that JSX.Element is redundant here?

Reply:
I'm pretty sure it's simply wrong. But again: this is why we need tests. This is probably accepting `{ component: <div /> }` when you want to accept `{ component: "div" }`.